src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs

```diff
@@ -367,6 +374,20 @@ private bool EvaluateFunctionBody(IPythonFunctionType fn)
             return parameters;
         }
 
+        private void ValidateParameters(FunctionDefinition fd) {
+            var duplicates = fd.Parameters.Where(p => !string.IsNullOrEmpty(p.Name)).GroupBy(p => p.Name).Where(g => g.Count() > 1).Select(y => y.Key);
```

Comment:
Could anyone give me a nicer syntax for this? 
I'm trying to find all duplicate names in ```fd.Parameters```

Reply:
I don't think there's a nicer way off the top of my head. Maybe `fd.Parameters.GroupBy(p => p.Name).SelectMany(g => g.Skip(1)).Distinct()`, but I'm not convinced that's better.

Reply:
yeah and we have to add the string null empty check on top of that too, so maybe this is the best.